org.opentosca.bus/org.opentosca.bus.management.api.soaphttp/src/main/java/org/opentosca/bus/management/api/soaphttp/processor/RequestProcessor.java

```diff
@@ -119,8 +123,21 @@ public void process(final Exchange exchange) throws Exception {
             final List<ResolvedDeploymentArtifact> resolvedDAs = new ArrayList<>();
             if (nodeTemplateID != null) {
                 final Csar csar = csarStorage.findById(new CsarId(csarIDString));
+                final TServiceTemplate serviceTemplate = ToscaEngine.resolveServiceTemplate(csar, serviceTemplateID);
+
                 final TNodeTemplate nodeTemplate = ToscaEngine.resolveNodeTemplate(csar, serviceTemplateID, nodeTemplateID);
 
+                if (Types.openStackTrainNodeType.getLocalPart().equals(nodeTemplateID)) {
```

Comment:
Please check the Node Type of the nodeTemplate, the ID might be different.

Can you maybe only check the name without the version and using starts with?

Reply:
Here it would be better to check with nodeTemplate.getType(), and instead of a startWith check we could check via type hierarchy? an "abstract" openstack nodetype and the specific one inherit from it? does that make sense?